mock-connector/src/main/scala/it/agilelab/darwin/connector/mock/MockConnector.scala

```diff
@@ -1,32 +1,92 @@
 package it.agilelab.darwin.connector.mock
 
 import com.typesafe.config.Config
-import it.agilelab.darwin.common.Connector
-import org.apache.avro.Schema
+import it.agilelab.darwin.common.compat._
+import it.agilelab.darwin.common.{Connector, Logging, SchemaReader}
+import org.apache.avro.{Schema, SchemaNormalization}
 
 import scala.collection.mutable
 
-class MockConnector(config: Config) extends Connector {
+class MockConnectorException(msg: String, t: Option[Throwable]) extends RuntimeException(msg) {
+  def this(msg: String) = this(msg, None)
 
-  val table: mutable.Map[Long, Schema] = {
-    val alone = parseResource("test/MockClassAlone.avsc")
-    val parent = parseResource("test/MockClassParent.avsc")
-    mutable.Map(0L -> alone, 1L -> parent)
+  def this(t: Throwable) = this(t.getMessage, Some(t))
+
+  override def getCause: Throwable = t match {
+    case Some(value) => value
+    case None => super.getCause
+  }
+}
+
+class MockConnector(config: Config) extends Connector with Logging {
```

Comment:
I think that since this class improved so much, it is a waste to use it only for test purposes, and it could be an actual file-based connector (maybe changing its name to something like FileConnector). I cannot think of an actual use case of this connector in a real world environment, but maybe someone will in the future :)

Reply:
Memory connector sounds good to me, I already thought about it while refactoring. What do you think of MemoryConnector?

Reply:
I like it, if you want an alternative it could be InMemoryConnector